apps/rss/rss.py

```diff
@@ -0,0 +1,180 @@
+        to_send = "I already track that url for you.."
+        publish(src, user, to_send)
+    else:
+        MEMORY[ident][alias] = { 'src'           : src,
+                                 'user'          : user,
+                                 'url'           : url,
+                                 'last_published': "" }
+
+    write_memory()
+    update()
+
+
+def del_rss(src, user, alias):
+    in_channel = src.split('.')[1] != src.split('.')[3]
+    if in_channel: ident = src
+    else: ident = user
+
+    if not MEMORY.has_key(ident):
+        to_send = "I don't even know you."
+    elif MEMORY[ident].has_key(alias):
+        MEMORY[ident].pop(alias)
+        write_memory()
+        to_send = "I won't track %s anymore." % (alias)
+    else:
+        to_send = "%s? .. What's that supposed to be?!" % (alias)
+
+    publish(src, user, to_send)
+
+def list_rss(src, user):
+    in_channel = src.split('.')[1] != src.split('.')[3]
+    if in_channel: ident = src
+    else: ident = user
+
+    if not MEMORY.has_key(ident):
+        to_send = "I don't track any RSS feeds."
+    elif [v['url'] for v in MEMORY[ident].values()] == []:
+        to_send = "I don't track any RSS feeds."
+    else:
+        to_send = "I track: %s" % (", ".join(MEMORY[ident].keys()))
+
+    publish(src, user, to_send)
+
+def rss_check(url):
+    try:
+        feed = feedparser.parse(url)
+        assert feed["bozo"] != 1 # bozo == 1 means, that the feed wasn't parsable.
+        assert feed.entries[0].title
+        assert feed.entries[0].link
+
+        res = feed
+    except:
+        res = False
+
+    return res
+
+def update():
+    for ident in MEMORY.keys():
+        for alias in MEMORY[ident].keys():
+            feed = rss_check(MEMORY[ident][alias]['url'])
+            src = MEMORY[ident][alias]['src']
+            user = MEMORY[ident][alias]['user']
+
+            if not feed:
+                to_send = "Uh.. I couldn't get %s for you." % (alias)
+                publish(src, user, to_send)
+                MEMORY[ident][alias]['last_published'] = to_send
+                continue
+
+            link = feed.entries[0].link
+            link = re.findall(r'.{1,400}', link) # TODO: wat?
+            link = "\n".join(link)
+
+            to_send = "%s - %s - %s" % (alias, feed.entries[0].title, link)
+            if to_send == MEMORY[ident][alias]['last_published']:
+                continue
+            else:
+                publish(src, user, to_send)
+                MEMORY[ident][alias]['last_published'] = to_send
+
+    write_memory()
+
+
+class FeedUpdater(Thread):
+    def __init__(self):
+        Thread.__init__(self)
+        self.daemon = True
+
+    def run(self):
+        while True:
+            log.debug("Updating Feeds")
+            update()
+            time.sleep(3*60)
+
+
+with edi.Manager(name="RSS", descr="RSS feed client.") as e:
+
+    @edi.edi_cmd(e, "sub",
+                 args="TEXT",
```

Comment:
Not yet used anywhere else, but I think using args=URL would be beneficial later